src/v/kafka/server/handlers/fetch.cc

```diff
@@ -456,95 +449,138 @@ class parallel_fetch_plan_executor final : public fetch_plan_executor::impl {
+     * the partitions already read and move to the end of iteration order
+     */
+    if (
+      session_ctx.is_sessionless()
+      || (session_ctx.is_full_fetch() && initial_fetch)) {
+        std::for_each(
+          request.cbegin(),
+          request.cend(),
+          [f = std::forward<Func>(f)](
+            const fetch_request::const_iterator::value_type& p) {
+              auto& part = *p.partition;
+              f(fetch_session_partition{
+                .topic_partition = {p.topic->name, part.partition_index},
+                .max_bytes = part.max_bytes,
+                .fetch_offset = part.fetch_offset,
+              });
+          });
+    } else {
+        std::for_each(
+          session_ctx.session()->partitions().cbegin_insertion_order(),
+          session_ctx.session()->partitions().cend_insertion_order(),
+          std::forward<Func>(f));
+    }
+}
+
 class simple_fetch_planner final : public fetch_planner::impl {
     fetch_plan create_plan(op_context& octx) final {
         fetch_plan plan(ss::smp::count);
         auto resp_it = octx.response_begin();
         auto bytes_left_in_plan = octx.bytes_left;
+
+        plan.reserve_from_partition_count(octx.fetch_partition_count());
+
         /**
          * group fetch requests by shard
          */
-        octx.for_each_fetch_partition(
-          [&resp_it, &octx, &plan, &bytes_left_in_plan](
-            const fetch_session_partition& fp) {
-              // if this is not an initial fetch we are allowed to skip
-              // partions that aleready have an error or we have enough data
-              if (!octx.initial_fetch) {
-                  bool has_enough_data = !resp_it->empty()
-                                         && octx.over_min_bytes();
-
-                  if (resp_it->has_error() || has_enough_data) {
-                      ++resp_it;
-                      return;
-                  }
-              }
-              /**
-               * if not authorized do not include into a plan
-               */
-              if (!octx.rctx.authorized(
-                    security::acl_operation::read, fp.topic)) {
-                  resp_it->set(make_partition_response_error(
-                    fp.partition, error_code::topic_authorization_failed));
-                  ++resp_it;
-                  return;
-              }
-
-              auto ntp = model::ntp(
-                model::kafka_namespace, fp.topic, fp.partition);
-
-              // there is given partition in topic metadata, return
-              // unknown_topic_or_partition error
-              if (unlikely(!octx.rctx.metadata_cache().contains(ntp))) {
-                  resp_it->set(make_partition_response_error(
-                    fp.partition, error_code::unknown_topic_or_partition));
-                  ++resp_it;
-                  return;
-              }
+        octx.for_each_fetch_partition([&resp_it,
```

Comment:
I do not know why clang-format wanted to reformat this call and lambda, but I could not convince it not to. It has changed the indentation from 6 spaces to 4. Probably best to review with "ignore whitespace".

Reply:
oof, this reformat is rough on the brain

Reply:
I actually struggled with clang-format a while trying to understand why it was doing this: I think some change of complexity inside the lamdba triggers it to somehow change the overall lamba format and indent level. Sorry.